projects/batfish/src/main/java/org/batfish/representation/juniper/JuniperConfiguration.java

```diff
@@ -247,6 +258,28 @@ private BgpProcess createBgpProcess(RoutingInstance routingInstance) {
       IpBgpGroup ig = e.getValue();
       BgpNeighbor neighbor = new BgpNeighbor(ip, _c);
       neighbor.setVrf(vrfName);
+      String authKeyChainName = ig.getAuthKeyChainName();
+      if (ig.getAuthKeyChainName() != null) {
+        if (!_c.getAuthenticationKeyChains().containsKey(authKeyChainName)) {
+          _w.redFlag(
```

Comment:
Don't use `_w.redFlag` here. Use `VendorConfiguration.undefined()` instead. You will need a line number to refer to with that call, so you may want to store that as a field like `authenticationKeyChainNameLine`.